### DISCUSSION TITLE
ModelRenderable with ChromaKey

### DISCUSSION THREAD

**Discussion by caiocs** [sceneview/sceneform-android#30] [Q&A]
Hello,

I used to use the SceneForm 1.15.0 and works fine with the example chromakeyvideo. They used a video_screen.obj and a material.mat, this generate a .sfb file and with that I could create a Model to show my video with chroma.

The .sfb file doesn't work any more with the SceneForm 1.16.0+. Any one has any sample how put the chromakeyvideo example to works on the the 1.18.0+?

**Comment by caiocs** [comment]
Here is the example how I used to load a ModelRenderable to insert my video with chroma, how can I do the same thing with a glb model and a external material to use the parameters inside this material (chroma)?
`ModelRenderable
                .builder()
                .setSource(this, Uri.parse("video_screen.sfb"))
                .build()
                .thenAccept(modelRenderable -> {
                    modelRenderable.getMaterial().setExternalTexture("videoTexture",
                            texture);
                    modelRenderable.getMaterial().setFloat4("keyColor",
                            new Color(0.1843f, 1f, 0.098f));
                    modelRenderable.getMaterial().setBoolean("disableChromaKey", !isChroma);
                    renderable = modelRenderable;
                });`

**Comment by ThomasGorisse** [comment]
If you can wait a little, since there are some others requests on it. I'll soon put a sample.
If not, you can make it like this :
- Convert the plane model obj to glb or create new one using blender or build it programmatically with ‘VertexBuffer‘ 
- Generate the .filamat from the .mat file using Filament Tool
- Load glb model and apply the external texture from your MediaPlayer
- Share your code and files here ;-)

**Reply by caiocs** [reply]
HI @ThomasGorisse,

How can I set the parameters from my material if I'm changing the whole material like a external texture... today my external texture it's where I put my media player. 
`ModelRenderable .builder() .setSource(this, Uri.parse("video_screen.sfb")) 
   .build() 
   .thenAccept(modelRenderable -> { 
      modelRenderable.getMaterial().setExternalTexture("videoTexture", texture);
      modelRenderable.getMaterial().setFloat4("keyColor", new Color(0.1843f, 1f, 0.098f)); 
      modelRenderable.getMaterial().setBoolean("disableChromaKey", !isChroma); 
      renderable = modelRenderable; 
   });`
The file video_screen.sfb it's the object that I use to play the video, the material that I have has the parameters: videoTexture (Used to play the video), keyColor (The color of the chroma) and disableChromaKey (Used to disable chroma when it's not needed). 

I tried to convert the video_screen.obj to video_screen.glb, but how can I load a material file inside the ModelRenderable to be able to set the parameters videoTexture, keyColor and disableChromaKey?

**Comment by ThomasGorisse** [comment]
The 1.18.8 release solves some Material issues and brings a Video Texture sample
[Release 1.18.8 - Video Texture + SceneView Background](https://github.com/ThomasGorisse/sceneform-android-sdk/releases/tag/v1.18.8)

**Reply by caiocs** [reply]
I'm getting this error when trying to get the version 1.18.8:


`Could not GET 'https://jitpack.io/com/gorisse/thomas/sceneform/sceneform/1.18.8/sceneform-1.18.8.pom'. Received status code 401 from server: Unauthorized`

**Comment by ThomasGorisse** [comment]
For the snapshot, the package name is not the same :
[https://thomasgorisse.github.io/sceneform-android-sdk/dependencies](https://thomasgorisse.github.io/sceneform-android-sdk/dependencies)

**Reply by caiocs** [reply]
Error: 
`Could not resolve all files for configuration ':echoarlib:debugCompileClasspath'.
   > Could not find com.github.ThomasGorisse.sceneform-android-sdk:sceneform:-SNAPSHOT.`

build.gradle:
`buildscript {
    repositories {
        jcenter()
        google()
        maven { url 'https://jitpack.io' }
    }
    dependencies {
        classpath 'com.android.tools.build:gradle:4.1.0'
        classpath 'com.google.ar.sceneform:plugin:1.15.0'
        // NOTE: Do not place your application dependencies here; they belong
        // in the individual module build.gradle files
    }
}

allprojects {
    repositories {
        jcenter()
        google()
        maven { url 'https://jitpack.io' }
    }
}`

implementation on app/build.gradle:
`implementation 'com.github.ThomasGorisse.sceneform-android-sdk:sceneform:-SNAPSHOT'`

**Comment by ThomasGorisse** [comment]
> For the snapshot, the package name is not the same :
> [https://thomasgorisse.github.io/sceneform-android-sdk/dependencies](https://thomasgorisse.github.io/sceneform-android-sdk/dependencies)

Thanks for the report.
It looks like an error in the last commit. I'll fix it quick.

**Comment by ThomasGorisse** [comment]
@caiocs I pushed an update.
Can you try again removing after cleaning the gradle cache ?

**Reply by caiocs** [reply]
Got a new error:

'Could not resolve all artifacts for configuration ':app:debugRuntimeClasspath'.
   > Could not find com.github.ThomasGorisse.sceneform-android-sdk:sceneform:-SNAPSHOT.
     Searched in the following locations:
       - https://jcenter.bintray.com/com/github/ThomasGorisse/sceneform-android-sdk/sceneform/-SNAPSHOT/maven-metadata.xml
       - https://jcenter.bintray.com/com/github/ThomasGorisse/sceneform-android-sdk/sceneform/-SNAPSHOT/sceneform--SNAPSHOT.pom
       - https://dl.google.com/dl/android/maven2/com/github/ThomasGorisse/sceneform-android-sdk/sceneform/-SNAPSHOT/maven-metadata.xml
       - https://dl.google.com/dl/android/maven2/com/github/ThomasGorisse/sceneform-android-sdk/sceneform/-SNAPSHOT/sceneform--SNAPSHOT.pom
       - https://jitpack.io/com/github/ThomasGorisse/sceneform-android-sdk/sceneform/-SNAPSHOT/maven-metadata.xml
       - https://jitpack.io/com/github/ThomasGorisse/sceneform-android-sdk/sceneform/-SNAPSHOT/sceneform--v1.18.7-g503a4ad-10.pom'
       
I already disabled the offline mode from Gradle.

**Reply by caiocs** [reply]
I downloaded the code on git hub and imported to my project and everything works fine. The problem it's to try import direct from git hub or JCenter.

**Comment by ThomasGorisse** [comment]
The problem comes from jitpack old cached versions.
Now that the 1.18.8 is out, you can just use the jCenter artifact by removing the `maven { url 'https://jitpack.io' }` lines
And using this dependency :
```
dependencies {
     implementation("com.gorisse.thomas.sceneform:sceneform:1.18.8")
}
```

**Reply by caiocs** [reply]
Hi @ThomasGorisse , 

I did that and doesn't work. I was at the bintray page and find the the url 'https://dl.bintray.com/thomasgorisse/maven/' on the top right of your page, putted this in my build.gradle and works.

maven { url 'https://dl.bintray.com/thomasgorisse/maven' }

if I put only jcenter the app cannot found the project.

**Comment by ThomasGorisse** [comment]
Fixed with mavenCentral() : 2494f6eeae57cc610ecb8bea01812754943f61d3
